lib/take_elevated.lua

```diff
@@ -48,14 +47,15 @@ local function calculateNewBucketContent(current, tokens_per_ms, bucket_size, cu
 end
 
 local function takeERLQuota(erl_quota_key, erl_quota_amount, erl_quota_expiration_epoch)
-    local erl_quota = erl_quota_amount
+    local erl_quota = 0
     local get_quota_result = redis.call('GET', erl_quota_key)
     if type(get_quota_result) == 'string' then
         erl_quota = tonumber(get_quota_result)
     end
 
-    if erl_quota > 0 then
-        redis.call('SET', erl_quota_key, erl_quota -1, 'PXAT', string.format('%.0f', erl_quota_expiration_epoch))
+    if erl_quota < erl_quota_amount then
+        erl_quota = erl_quota + 1
+        redis.call('SET', erl_quota_key, erl_quota, 'PXAT', string.format('%.0f', erl_quota_expiration_epoch))
     end
     return erl_quota
 end
```

Comment:
I find mutable variables hard to follow, could we change this to have explicit returns for each case? Something like the following;

```suggestion
local function takeERLQuota(erl_quota_key, erl_quota_amount, erl_quota_expiration_epoch)
    if erl_quota_amount <= 0 then
        return 0
    end

    local get_quota_result = redis.call('GET', erl_quota_key)
    if type(get_quota_result) ~= 'string' then
        redis.call('SET', erl_quota_key, 1, 'PXAT', string.format('%.0f', erl_quota_expiration_epoch))
        return 1
    end

    local erl_quota = tonumber(get_quota_result)

    if erl_quota >= erl_quota_amount then
        return erl_quota
    end

    local incremented_erl_quota = erl_quota + 1
    redis.call('SET', erl_quota_key, incremented_erl_quota, 'PXAT', string.format('%.0f', erl_quota_expiration_epoch))
    return incremented_erl_quota
end
```

Reply:
Done